src/lib.rs

```diff
@@ -1721,6 +1721,87 @@ pub fn strongly_connected_components(
         .collect()
 }
 
+/// Return the first cycle encountered during DFS of a given PyDiGraph,
+/// empty list is returned if no cycle is found
+///
+/// :param PyDiGraph graph: The graph to find the cycle in
+/// :param int root: Optional index for starting node for cycle
+///
+/// :returns: A list describing the cycle. The index of node ids which
+///     forms a cycle (loop) in the input graph
+/// :rtype: list
+#[pyfunction]
+#[text_signature = "(graph, /, root=None)"]
+pub fn digraph_find_cycle(
+    graph: &digraph::PyDiGraph,
+    root: Option<usize>,
+) -> Vec<(usize, usize)> {
+    let root_node = root;
+    let mut graph_nodes: HashSet<NodeIndex> =
+        graph.graph.node_indices().collect();
+    let mut cycle: Vec<(usize, usize)> = Vec::new();
+    let temp_value: NodeIndex;
+    // If root_node is not set get an arbitrary node from the set of graph
+    // nodes we've not "examined"
+    let root_index = match root_node {
```

Comment:
```suggestion
    let root_index = match root {
```

or really after we change `root` -> `source` above

```suggestion
    let root_index = match source {
```